lib/gems/pending/util/miq-password.rb

```diff
@@ -185,16 +186,79 @@ def self.ez_load(filename, recent = true)
     if !File.exist?(filename)
       nil
     elsif recent
-      EzCrypto::Key.load(filename)
+      params = YAML.load_file(filename)
+      Key.new(*params.values_at(:algorithm, :key, :iv))
     else
       params = YAML.load_file(filename)
-      CryptString.new(nil, params[:algorithm], params[:key], params[:iv])
+      algorithm, key, iv = params.values_at(:algorithm, :key, :iv)
+      Key.new(algorithm, key && Base64.encode64(key), iv && Base64.encode64(iv))
     end
   end
 
   def self.extract_erb_encrypted_value(value)
     return $1 if value =~ /\A<%= (?:MiqPassword|DB_PASSWORD)\.decrypt\(['"]([^'"]+)['"]\) %>\Z/
   end
+
+  class Key
+    GENERATED_KEY_SIZE = 32
+
+    def self.generate_key(password = nil, salt = nil)
+      password ||= OpenSSL::Random.random_bytes(GENERATED_KEY_SIZE)
+      Base64.encode64(Digest::SHA256.digest("#{password}#{salt}")[0, GENERATED_KEY_SIZE]).chomp
+    end
+
+    def initialize(algorithm = nil, key = nil, iv = nil)
+      @algorithm = algorithm || "aes-256-cbc"
+      @key       = key || generate_key
+      @raw_key   = Base64.decode64(@key)
+      @iv        = iv
```

Comment:
Although we don't have code that uses it at the moment, it felt better to keep it in the API for consistency.  We can always remove in a followup PR, I think.